pkg/object/sftp.go

```diff
@@ -380,7 +380,17 @@ func (f *sftpStore) ListAll(prefix, marker string) (<-chan *Object, error) {
 }
 
 func newSftp(endpoint, user, pass string) (ObjectStorage, error) {
-	parts := strings.Split(endpoint, ":")
+	var hostEnd int
+	// IPv6 address like [fe80::148a:d031:823a:fff]
+	if strings.HasPrefix(endpoint, "[") {
+		hostEnd = strings.Index(endpoint, "]:") + 1
+	} else { // IPv4 address or host name
+		hostEnd = strings.Index(endpoint, ":")
+	}
+	if hostEnd <= 0 {
+		return nil, fmt.Errorf("unable to parse host from endpoint: %s", endpoint)
+	}
+	parts := strings.Split(endpoint[hostEnd:], ":")
```

Comment:
Should we use net.SplitHostPort? https://golang.org/pkg/net/#SplitHostPort

Reply:
The port may not exist like: `[IPv6]:/dir` and we use `22` as default.